pkg/store/bucket.go

```diff
@@ -724,6 +746,19 @@ func blockSeries(
 				break
 			}
 
+			mask := false
+			for _, ts := range validTombstones {
+				if meta.MaxTime > ts.MinTime && meta.MaxTime <= ts.MaxTime {
```

Comment:
We had some method somewhere for checking if intervals of tombstones overlaps or not, maybe we can use that.

Just make sure add TODO to rewrite chunk on partial overlap with tombstone.